src/custom-element.js

```diff
@@ -963,6 +969,13 @@ function createBaseCustomElementClass(win) {
         const ampdocService = Services.ampdocServiceFor(win);
         const ampdoc = ampdocService.getAmpDoc(this);
         this.ampdoc_ = ampdoc;
+
+        // Make sure that the ampdoc has already been stubbed.
+        if (!docInitializedMap.has(ampdoc)) {
+          docInitializedMap.set(ampdoc);
```

Comment:
Nit: to follow the types
```suggestion
          docInitializedMap.set(ampdoc, true);
```